python/mxnet/test_utils.py

```diff
@@ -1407,12 +1410,15 @@ def download(url, fname=None, dirname=None, overwrite=False):
                 if exc.errno != errno.EEXIST:
                     raise OSError('failed to create ' + dirname)
 
-    if not overwrite and os.path.exists(fname):
+    if not overwrite and os.path.exists(fname) and not version_tag:
         logging.info("%s exists, skipping download", fname)
         return fname
 
     r = requests.get(url, stream=True)
     assert r.status_code == 200, "failed to open %s" % url
+    if version_tag and r.headers['ETag'] != version_tag:
+        logging.info("The version tag of the file does not match the expected version. "
+                     + "Proceeding with the file download...")
```

Comment:
why do you have to download if they don't match

Reply:
the download method was already present in the test_utils module and in use by other modules in the project. I did not want the eTag mismatch to fail any existing tests.
Also I used this as a reference - https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/utils.py#L216-L220

Reply:
Oh right, good point. We should rather fail if the file is not the one we are expecting.